src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -283,6 +297,21 @@ public ISymbolNode HelperEntrypoint(HelperEntrypoint entrypoint)
             return symbol;
         }
 
+        private TypeDesc _systemICastableType;
+
+        public TypeDesc ICastableInterface
```

Comment:
Even if I squint really hard, this doesn't have any connection with nodes or dependency analysis. NodeFactory shouldn't become a dumping ground for "things that are needed in a node somewhere".

This should go on CompilerTypeSystemContext or a place like that.